redash/handlers/widgets.py

```diff
@@ -52,10 +52,20 @@ def post(self):
 
 class WidgetResource(BaseResource):
     @require_permission('edit_dashboard')
+    def post(self, widget_id):
+        widget = models.Widget.get_by_id_and_org(widget_id, self.current_org)
+        require_admin_or_owner(widget.dashboard.user_id)
+        widget_properties = request.get_json(force=True)
+        widget.text = widget_properties['text']
+        widget.width = widget_properties['width']
```

Comment:
Also I would add a comment here mentioning that it currently handles Text Box widgets only.